trace/run.go

```diff
@@ -8,20 +8,27 @@ import (
 	akihttp "github.com/akitasoftware/akita-libs/akinet/http"
 )
 
-func Collect(stop <-chan struct{}, intf, bpfFilter string, proc Collector) error {
+func Collect(stop <-chan struct{}, intf, bpfFilter string, proc Collector, packetCount PacketCountConsumer) error {
 	defer proc.Close()
 
 	facts := []akinet.TCPParserFactory{
 		akihttp.NewHTTPRequestParserFactory(),
 		akihttp.NewHTTPResponseParserFactory(),
 	}
+
 	parser := col.NewNetworkTrafficParser()
+
+	if packetCount != nil {
+		parser.InstallObserver(CountTcpPackets(intf, packetCount))
+	}
```

Comment:
Why not roll this into the constructor?

Reply:
I wasn't sure how many times NewNetworkTrafficParser was being used.  (The answer is it's only 4, so it would have been reasonable.)

The idiomatic issue I have is that I hate adding nil's (even though I did it elsewhere.)  It is much better, in my opinion, to have either

```
newObj := Blah()
newObj.SetUpFoo()
```

or

newObj := Blah( &BlahConfig {
  Foo: fooConfig,
} )
```

than

```
newObj := Blah( nil, nil, fooConfig, nil )
```

But, I didn't have the guts to refactor NewBackendCollector into the latter form.